### ISSUE TITLE
Unknown CMake command "SHLOMIF_COMMON_SETUP"

### ISSUE THREAD

**Issue by ShiSanChuan** [thirtythreeforty/businesscard-linux#15]
hello,I compile according to the tutorial, after executing the "make" command, it appears "Unknown CMake command "SHLOMIF_COMMON_SETUP"

`
/usr/bin/make -j1 O=/home/shisanchuan/C++work/businesscard-linux/buildroot/output HOSTCC="/usr/bin/gcc" HOSTCXX="/usr/bin/g++" syncconfig
>>> host-fortune-mod 2.10.0 Configuring
(mkdir -p /home/shisanchuan/C++work/businesscard-linux/buildroot/output/build/host-fortune-mod-2.10.0/fortune-mod/buildroot-build && cd /home/shisanchuan/C++work/businesscard-linux/buildroot/output/build/host-fortune-mod-2.10.0/fortune-mod/buildroot-build && rm -f CMakeCache.txt && PATH="/home/shisanchuan/C++work/businesscard-linux/buildroot/output/host/bin:/home/shisanchuan/C++work/businesscard-linux/buildroot/output/host/sbin:/usr/local/cuda-9.0/bin:/usr/local/MATLAB/R2017a/bin:/usr/local/cuda-9.0/bin:/home/shisanchuan/bin:/home/shisanchuan/.local/bin:/usr/local/bin/cmake:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:AVA_HOME/bin:/snap/bin:/opt/lingo13/bin/linux64:/home/shisanchuan/Qt5.11.1/5.11.1/gcc_64/bin" PKG_CONFIG="/home/shisanchuan/C++work/businesscard-linux/buildroot/output/host/bin/pkg-config" PKG_CONFIG_SYSROOT_DIR="/" PKG_CONFIG_LIBDIR="/home/shisanchuan/C++work/businesscard-linux/buildroot/output/host/lib/pkgconfig:/home/shisanchuan/C++work/businesscard-linux/buildroot/output/host/share/pkgconfig" PKG_CONFIG_ALLOW_SYSTEM_CFLAGS=1 PKG_CONFIG_ALLOW_SYSTEM_LIBS=1  /usr/local/bin/cmake /home/shisanchuan/C++work/businesscard-linux/buildroot/output/build/host-fortune-mod-2.10.0/fortune-mod -DCMAKE_INSTALL_SO_NO_EXE=0 -DCMAKE_FIND_ROOT_PATH="/home/shisanchuan/C++work/businesscard-linux/buildroot/output/host" -DCMAKE_FIND_ROOT_PATH_MODE_PROGRAM="BOTH" -DCMAKE_FIND_ROOT_PATH_MODE_LIBRARY="BOTH" -DCMAKE_FIND_ROOT_PATH_MODE_INCLUDE="BOTH" -DCMAKE_INSTALL_PREFIX="/home/shisanchuan/C++work/businesscard-linux/buildroot/output/host" -DCMAKE_C_FLAGS="-O2 -I/home/shisanchuan/C++work/businesscard-linux/buildroot/output/host/include" -DCMAKE_CXX_FLAGS="-O2 -I/home/shisanchuan/C++work/businesscard-linux/buildroot/output/host/include" -DCMAKE_EXE_LINKER_FLAGS="-L/home/shisanchuan/C++work/businesscard-linux/buildroot/output/host/lib -Wl,-rpath,/home/shisanchuan/C++work/businesscard-linux/buildroot/output/host/lib" -DCMAKE_SHARED_LINKER_FLAGS="-L/home/shisanchuan/C++work/businesscard-linux/buildroot/output/host/lib -Wl,-rpath,/home/shisanchuan/C++work/businesscard-linux/buildroot/output/host/lib" -DCMAKE_ASM_COMPILER="/usr/bin/as" -DCMAKE_C_COMPILER="/home/shisanchuan/C++work/businesscard-linux/buildroot/output/host/bin/ccache" -DCMAKE_CXX_COMPILER="/home/shisanchuan/C++work/businesscard-linux/buildroot/output/host/bin/ccache"  -DCMAKE_C_COMPILER_ARG1="/usr/bin/gcc" -DCMAKE_CXX_COMPILER_ARG1="/usr/bin/g++"  -DCMAKE_COLOR_MAKEFILE=OFF -DBUILD_DOC=OFF -DBUILD_DOCS=OFF -DBUILD_EXAMPLE=OFF -DBUILD_EXAMPLES=OFF -DBUILD_TEST=OFF -DBUILD_TESTS=OFF -DBUILD_TESTING=OFF   )
-- The C compiler identification is GNU 5.4.0
-- The CXX compiler identification is GNU 5.4.0
-- Check for working C compiler: /home/shisanchuan/C++work/businesscard-linux/buildroot/output/host/bin/ccache
-- Check for working C compiler: /home/shisanchuan/C++work/businesscard-linux/buildroot/output/host/bin/ccache -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /home/shisanchuan/C++work/businesscard-linux/buildroot/output/host/bin/ccache
-- Check for working CXX compiler: /home/shisanchuan/C++work/businesscard-linux/buildroot/output/host/bin/ccache -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at CMakeLists.txt:4 (SHLOMIF_COMMON_SETUP):
  Unknown CMake command "SHLOMIF_COMMON_SETUP".
-- Configuring incomplete, errors occurred!
See also "/home/shisanchuan/C++work/businesscard-linux/buildroot/output/build/host-fortune-mod-2.10.0/fortune-mod/buildroot-build/CMakeFiles/CMakeOutput.log".
package/pkg-generic.mk:228: recipe for target '/home/shisanchuan/C++work/businesscard-linux/buildroot/output/build/host-fortune-mod-2.10.0/.stamp_configured' failed
make[1]: *** [/home/shisanchuan/C++work/businesscard-linux/buildroot/output/build/host-fortune-mod-2.10.0/.stamp_configured] Error 1
Makefile:84: recipe for target '_all' failed
make: *** [_all] Error 2
`

**Comment by thirtythreeforty** [comment]
You are not the first person to get this error.  I am unable to reproduce it, however.  Would you mind telling me what operating system you are running on your workstation?

**Comment by solhuebner** [comment]
This seems to be an issue with the FORTUNE_MOD so I removed the lines in the Config.in and configs/thirtythreeforty_businesscard_defconfig files and it compiled fine. I am running CentOS 7. @thirtythreeforty thanks for this project btw!

**Comment by ShiSanChuan** [comment]
> You are not the first person to get this error. I am unable to reproduce it, however. Would you mind telling me what operating system you are running on your workstation?

I am running the Ubantu system

**Comment by ShiSanChuan** [comment]
> This seems to be an issue with the FORTUNE_MOD so I removed the lines in the Config.in and configs/thirtythreeforty_businesscard_defconfig files and it compiled fine. I am running CentOS 7. @thirtythreeforty thanks for this project btw!

Xièxiè, wǒ qùdiào xuǎnxiàng hòu chénggōng biānyì chūlái de, wǒ zhǔnbèi zài jiē xiàlái yīduàn shíjiān qù shì shì zhège jìngxiàng
35/5000
ohhhhhhh, thanks, I successfully compiled it after removing the options, and I plan to try this mirror in the next period

**Comment by ShiSanChuan** [comment]
> This seems to be an issue with the FORTUNE_MOD so I removed the lines in the Config.in and configs/thirtythreeforty_businesscard_defconfig files and it compiled fine. I am running CentOS 7. @thirtythreeforty thanks for this project btw!

Xièxiè, wǒ qùdiào xuǎnxiàng hòu chénggōng biānyì chūlái de, wǒ zhǔnbèi zài jiē xiàlái yīduàn shíjiān qù shì shì zhège jìngxiàng
35/5000
ohhhhhhh, thanks, I successfully compiled it after removing the options, and I plan to try this mirror in the next period

**Comment by thirtythreeforty** [comment]
Thanks for the host system info. Ideally the fortune package should compile for everyone, so I'll try to fix it if I can reproduce it in a CentOS container.

**Comment by baolintian** [comment]
> > This seems to be an issue with the FORTUNE_MOD so I removed the lines in the Config.in and configs/thirtythreeforty_businesscard_defconfig files and it compiled fine. I am running CentOS 7. @thirtythreeforty thanks for this project btw!
> 
> ohhhhhhh, thanks, I successfully compiled it after removing the options, and I plan to try this mirror in the next period

Excuse me, I also encountered the same problem but I don't know which lines should remove from the error information. Could you point out the specific lines or some key words?

**Comment by baolintian** [comment]
> > > This seems to be an issue with the FORTUNE_MOD so I removed the lines in the Config.in and configs/thirtythreeforty_businesscard_defconfig files and it compiled fine. I am running CentOS 7. @thirtythreeforty thanks for this project btw!
> > 
> > 
> > ohhhhhhh, thanks, I successfully compiled it after removing the options, and I plan to try this mirror in the next period
> 
> Excuse me, I also encountered the same problem but I don't know which lines should remove from the error information. Could you point out the specific lines or some key words?

YES! I also solved this problem. Remember delete the lines about ' FORTUNE_MOD' in 'Config.in' and 'configs/thirtythreeforty_businesscard_defconfig'. This two files at top directory.

**Comment by MichaelKirsch** [comment]
Had the same problem. I disabled all the games and ran apt upgrade on my host. now it compiles. maybe that will help others too

**Comment by Iscle** [comment]
Same error on Ubuntu 20.04 x86_64!

**Comment by mingusdenny** [comment]
I had the same issue. The location for the cmake file is 404.

Instead I downloaded from here:

https://github.com/shlomif/shlomif-cmake-modules

**Comment by ChangJiahong** [comment]
> I had the same issue. The location for the cmake file is 404.
> 
> Instead I downloaded from here:
> 
> https://github.com/shlomif/shlomif-cmake-modules

YES! I downloaded the corresponding Shlomif_Common.cmake from here (https://github.com/shlomif/shlomif-cmake-modules) and replaced it under the businesscard-linux/buildroot/output/build/fortune-mod-2.10.0/fortune-mod/cmake folder , I don't know what caused the Shlomif_Common.cmake under this folder to be empty, after replacing, it worked fine.
